rust/hyperlane-base/src/types/multisig.rs

```diff
@@ -62,14 +62,13 @@ impl MultisigCheckpointSyncer {
             // The highest viable checkpoint index is the minimum of the highest index
             // we (supposedly) have a quorum for, and the maximum index for which we can
             // generate a proof.
-            let mut index = std::cmp::min(*highest_quorum_index, maximum_index);
-            while index >= minimum_index {
+            let start_index = std::cmp::min(*highest_quorum_index, maximum_index);
+            for index in (minimum_index..=start_index).rev() {
```

Comment:
Do we know for a fact that `minimum_index <= start_index`, I don't see a guarantee of this but also not sure if it matters.

Reply:
No, good point, added the following:
```
            if (minimum_index > *start_index) {
                return Ok(None);
            }
```